lib/services/nodes-api-service.js

```diff
@@ -0,0 +1,96 @@
+// Copyright 2015, EMC, Inc.
+
+'use strict';
+
+var di = require('di');
+
+module.exports = nodeApiServiceFactory;
+di.annotate(nodeApiServiceFactory, new di.Provide('Http.Services.Api.Nodes'));
+di.annotate(nodeApiServiceFactory,
+    new di.Inject(
+        'Protocol.TaskGraphRunner',
+        'Services.Waterline',
+        'Errors',
+        'Logger',
+        '_',
+        'Promise'
+    )
+);
+function nodeApiServiceFactory(
+    taskGraphProtocol,
+    waterline,
+    Errors,
+    Logger,
+    _,
+    Promise
+) {
+    var logger = Logger.initialize(nodeApiServiceFactory);
+
+    function NodeApiService() {
```

Comment:
Good, glad we have a node api service. Just as a note it might make sense going forward for us to start moving the rest of logic in the node routes into here (switch/pdu discovery, etc.).